integration/simulation/simulation.go

```diff
@@ -42,11 +42,16 @@ func (s *Simulation) Start() {
 	timer := time.Now()
 	go s.TxInjector.Start()
 
+	stoppingDelay := 2 * time.Second
+
 	// Wait for the simulation time
-	time.Sleep(s.SimulationTime)
+	time.Sleep(s.SimulationTime - stoppingDelay)
```

Comment:
Does it need to account the stop delay here?

Reply:
if you want to run a 15s simulation, and you need to stop injection 13 seconds in, then you need to sleep 2 seconds afterwards

Reply:
it could be done differently, but I'd say this is good enough for now